OSXDemo/OCTConversationViewController.m

```diff
@@ -0,0 +1,264 @@
+//  objcTox
+//
+//  Created by Chuong Vu on 12/20/15.
+//  Copyright © 2015 dvor. All rights reserved.
+//
+
+#import "OCTConversationViewController.h"
+#import "RBQFetchedResultsController.h"
+#import "OCTSubmanagerObjects.h"
+#import "OCTSubmanagerChats.h"
+#import "OCTFriend.h"
+#import "OCTChat.h"
+#import "OCTMessageAbstract.h"
+#import "OCTSubmanagerUser.h"
+#import "OCTMessageText.h"
+
+static NSString *const kCellIdent = @"cellIdent";
+
+@interface OCTConversationViewController () <NSTableViewDataSource,
+                                             NSTableViewDelegate,
+                                             RBQFetchedResultsControllerDelegate, NSTextFieldDelegate>
+
+@property (weak) IBOutlet NSTableView *chatsViewController;
+@property (strong, nonatomic) OCTManager *manager;
+@property (strong, nonatomic) RBQFetchedResultsController *chatResultsController;
+@property (strong, nonatomic) RBQFetchedResultsController *conversationResultsController;
+@property (weak) IBOutlet NSTableView *chatsTableView;
+@property (weak) IBOutlet NSTableView *conversationTableView;
+
+@end
+
+@implementation OCTConversationViewController
+
+- (instancetype)initWithManager:(OCTManager *)manager
+{
+    self = [super init];
+
+    if (! self) {
+        return nil;
+    }
+
+    _manager = manager;
+
+    RBQFetchRequest *fetchRequest = [self.manager.objects
+                                     fetchRequestForType:OCTFetchRequestTypeChat
+                                           withPredicate:nil];
+
+    _chatResultsController = [[RBQFetchedResultsController alloc]
+                              initWithFetchRequest:fetchRequest
+                                sectionNameKeyPath:nil cacheName:nil];
+
+    _chatResultsController.delegate = self;
+    [_chatResultsController performFetch];
+
+    _conversationResultsController = [RBQFetchedResultsController new];
+    _conversationResultsController.delegate = self;
+
+    return self;
+}
+
+#pragma mark - Actions
+
+- (IBAction)deleteChatButtonPressed:(NSButton *)sender
+{
+    NSInteger selectedRow = self.chatsTableView.selectedRow;
+
+    if (selectedRow < 0) {
+        return;
+    }
+
+    NSIndexPath *path = [NSIndexPath indexPathForRow:selectedRow inSection:0];
+
+    OCTChat *chat = [self.chatResultsController objectAtIndexPath:path];
+
+    [self.manager.chats removeChatWithAllMessages:chat];
+}
+
+- (void)tableViewSelectionDidChange:(NSNotification *)notification
+{
+    if (notification.object != self.chatsTableView) {
+        return;
+    }
+
+    NSInteger selectedRow = self.chatsTableView.selectedRow;
+
+    if (selectedRow < 0) {
+        return;
+    }
+
+    NSIndexPath *path = [NSIndexPath indexPathForRow:selectedRow inSection:0];
+    OCTChat *chat = [self.chatResultsController objectAtIndexPath:path];
+
+    [self updateConversationControllerForChat:chat];
+
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self.conversationTableView reloadData];
+    });
```

Comment:
To ensure any UI stuff goes on the main thread. Though I guess it's not necessary needed here since this method is always under the main queue?